public/js/hotkey_binder/model/addShortcut.js

```diff
@@ -0,0 +1,86 @@
+//Assign the callback func for the keymap here
+
+const addShortcut = (keys, action) => {
+	let callback;
+	switch (action) {
+		case "New Project":
+			callback = newProject;
+			break;
+		case "Save Online":
+			callback = save;
+			break;
+		case "Save Offline":
+			callback = saveOffline;
+			break;
+		case "Download as Image":
+			callback = createSaveAsImgPrompt;
+			break;
+		case "Open Offline":
+			callback = createOpenLocalPrompt;
+			break;
+		case "New Circuit":
+			callback = newCircuit;
+			break;
+		case "Create Sub-circuit":
+			callback = createSubCircuitPrompt;
+			break;
+		case "Combinational Analysis":
+			callback = createCombinationalAnalysisPrompt;
+			break; //bug
+		case "Start Plot":
+			callback = startPlot;
+			break;
+		case "Direction Up":
+			callback = elementDirection('up');
+			break;
+		case "Direction Down":
+			callback = elementDirection('down');
+			break;
+		case "Direction Left":
+			callback = elementDirection('left');
+			break;
+		case "Direction Right":
+			callback = elementDirection('right');
+			break;
+		case "Insert Label":
+			callback = insertLabel;
+			break;
+		case "Label Direction Up":
+			callback = labelDirection('up');
+			break;
+		case "Label Direction Down":
+			callback = labelDirection('down');
+			break;
+		case "Label Direction Left":
+			callback = labelDirection('left');
+			break;
+		case "Label Direction Right":
+			callback = labelDirection('right');
+			break;
+		case "Move Element Up":
+			callback = moveElement('up');
+			break;
+		case "Move Element Down":
+			callback = moveElement('down');
+			break;
+		case "Move Element Left":
+			callback = moveElement('left');
+			break;
+		case "Move Element Right":
+			callback = moveElement('right');
+			break;
+		case "Hotkey Preference":
+			callback = openHotkey;
+			break;
+			
+		default:
+			callback = () => true;
+			break;
```

Comment:
Why default callback? Shouldn't it throw an error? or ignore.

Reply:
If new shortcuts are added correctly, it will never hit the default case, so far it doesn't, I just added it as code-climate suggested so. If you suggest I shall remove it.

Reply:
> 
> 
> CMD is working on Mac as required. I found two files which have `value.replace("Ctrl", "Meta")`, do we need to have them in 2 places?
> 
value.replace("Ctrl", "Meta")`: one for the shortcuts, one for restricted shortcuts, I could have have an util func for that..didn't....

Reply:
@abstrekt throw an error, its better to know something went wrong than nothing happening.